test/integration/consul-container/test/troubleshoot/troubleshoot_proxy_test.go

```diff
@@ -0,0 +1,73 @@
+package troubleshoot
+
+import (
+	"context"
+	"fmt"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	libservice "github.com/hashicorp/consul/test/integration/consul-container/libs/service"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/topology"
+	"github.com/hashicorp/consul/test/integration/consul-container/test/observability"
+	"github.com/stretchr/testify/require"
+	"io"
+	"testing"
+	"time"
+)
+
+func TestTroubleshootProxy_Success(t *testing.T) {
+
+	cluster, _, _ := topology.NewPeeringCluster(t, 1, &libcluster.BuildOptions{
+		Datacenter:           "dc1",
+		InjectAutoEncryption: true,
+	})
+
+	_, clientService := observability.CreateServices(t, cluster)
+
+	clientSidecar, ok := clientService.(*libservice.ConnectContainer)
+	require.True(t, ok)
+	_, port := clientSidecar.GetInternalAdminAddr()
+	// wait for envoy
+	time.Sleep(10 * time.Second)
+	_, outputReader, err := clientSidecar.Exec(context.Background(), []string{"consul", "troubleshoot", "proxy",
+		"-envoy-admin-endpoint", fmt.Sprintf("localhost:%v", port),
+		"-upstream-envoy-id", libservice.StaticServerServiceName})
+	require.NoError(t, err)
+	buf, err := io.ReadAll(outputReader)
```

Comment:
I think you can make the Exec returns the string so that we don't need line 34 in other places. Please refer to https://github.com/hashicorp/consul/blob/78a4b5fd6f812c1ba288aa0ad84de62c6f4a88d3/test/integration/consul-container/libs/cluster/container.go#L364-L366